### DISCUSSION TITLE
ResetPose service

### DISCUSSION THREAD

**Discussion by arielc-brillianetor** [iRobotEducation/create3_docs#81] [Questions] [answered]
Hi,
I was wondering if there is an example of how to use the ResetPos service. To my understanding it is supposed to set the position and orientation of the robot which is very important when setting the origin of a coordinate system. I tried running the following code but it didn't work:
ros2 service call /reset_pos irobot_create_msgs/srv/ResetPose "{position: {x: 1, y: 2, z: 0}, orientation: {x: 0, y: 0, z: 0, w: 1}}"

I get the following output:
Failed to populate field: 'ResetPose_Request' object has no attribute 'position'

If you can please assist me with this I would really appreciate it.
Thanks a lot

**Comment by jomoengineer** [comment]
Looking at the code for ResetPose, it appears it is looking for 'pose' to be past which in turns passes the parameters to 'geometry_msgs/Pos'
[https://github.com/iRobotEducation/irobot_create_msgs/blob/7d749943ae0054cadec97b5af13de8d2b7fe3d6c/srv/ResetPose.srv](url) 
Ex:
```
# This service reset the robot pose estimate to the provided one

geometry_msgs/Pose pose
---
```
Try adding 'pose' to the parameter list:
Ex:
```
ros2 service call /reset_pose irobot_create_msgs/srv/ResetPose "pose: {position: {x: 1, y: 2, z: 0}, orientation: {x: 0, y: 0, z: 0, w: 1}}"

```

**Reply by arielc-brillianetor** [reply]
thank you very much! I will try it and let you know.

**Reply by justinIRBT** [reply]
This service is not currently implemented on the robot.  The pose should reset if you restart the application.

**Reply by arielc-brillianetor** [reply]
So I can't set the pose of the robot? not sure what you mean when you say: "The pose should reset if you restart the application.".

**Reply by justinIRBT** [reply]
It resets to 0,0,0 (x,y,theta) when the application restarts.  There isn’t currently a way to have the robots internal coordinate system reset to a new position, but you could make your own positioner that builds off the robot position changes with a different starting position that you can manage, then have your application subscribe to that positioner

**Comment by scharbonneau8661** [comment]
Hi,

It is indeed not implemented but it should be according to the iRobot Create documentation (see the second service at the bottom of the page):
https://github.com/iRobotEducation/irobot_create_msgs

To reset the position of my robot, I currently use the RobotPower service to power off the robot (see last service at the bottom of the page):
https://github.com/iRobotEducation/irobot_create_msgs

It is definitely an overkill to have to power off the robot just to reset its position to zero but it works.

Is there a more elegant work around to reset the robot position?

Is there an ETA on when a new version of the robot firmware will include this missing capability?

Thank you and best regards,

Serge

**Reply by shamlian** [reply]
You don't need to power down the robot; you can just restart the application with a well-placed `curl` call; see https://iroboteducation.github.io/create3_docs/webserver/application/#restart-application .

**Reply by scharbonneau8661** [reply]
Unfortunately, it is not this simple since the restart of the application is performed by a web socket connection managed in JavaScript under the Restart Application html page.  Curl returns the content of the Restart Application page but does not execute the JavaScript since it can only be executed in a Web browser.  Is there another way to restart the application programmatically?

**Comment by shamlian** [comment, marked as answer]
This should be fixed in [G.3.1,](https://iroboteducation.github.io/create3_docs/releases/g_3_1/) which was released today. Please let us know if it works for you.

**Reply by scharbonneau8661** [reply]
It is fixed and it works perfectly.  Thank you for implementing the fix in the new version of the firmware and best regards.

**Reply by alsora** [reply]
Note that the current implementation of the `ResetPose` server always resets to the origin, i.e. it ignores the pose value that you may specify in the message.
